tower/src/make/make_infallible_service.rs

```diff
@@ -0,0 +1,174 @@
+use crate::sealed::Sealed;
+use pin_project_lite::pin_project;
+use std::future::Future;
+use std::{
+    convert::Infallible,
+    pin::Pin,
+    task::{Context, Poll},
+};
+use tower_service::Service;
+
+// TODO: documentation :)
+
+pub trait MakeInfallibleService<S, F, Request>: Sealed<(Request,)> {
+    fn make_infallible(self, err_into_ok: F) -> InfallibleService<S, F>;
+}
+
+impl<S, F, Request> MakeInfallibleService<S, F, Request> for S
+where
+    S: Service<Request>,
+{
+    fn make_infallible(self, err_into_ok: F) -> InfallibleService<S, F> {
+        InfallibleService {
+            service: self,
+            err_into_ok,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct InfallibleService<S, F> {
+    service: S,
+    err_into_ok: F,
+}
+
+impl<S, F> Clone for InfallibleService<S, F>
+where
+    S: Clone,
+    F: Clone,
+{
+    fn clone(&self) -> Self {
+        Self {
+            service: self.service.clone(),
+            err_into_ok: self.err_into_ok.clone(),
+        }
+    }
+}
+
+impl<R, S, F> Service<R> for InfallibleService<S, F>
+where
+    S: Service<R>,
+    S::Future: Send,
+    F: FnMut(S::Error) -> S::Response,
+    F: Clone,
+{
+    type Response = S::Response;
+    type Error = Infallible;
+    type Future = InfallibleCall<F, S::Future>;
+
+    fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        match self.service.poll_ready(cx) {
+            Poll::Pending => Poll::Pending,
+            Poll::Ready(Ok(())) => Poll::Ready(Ok(())),
+            Poll::Ready(Err(_)) => panic!("poll_ready must not return an error!"),
```

Comment:
Hm panicking here doesn't feel right 😕

axum already has https://docs.rs/axum/0.6.7/axum/error_handling/struct.HandleErrorLayer.html. I don't think it's appropriate to put in tower though. It doesn't discard services when poll_ready returns an error which ideally you should.

Reply:
I thought there is nothing more reasonable to do here than panicking, but yes I also struggled here.

So great, didn't notice `HandleErrorLayer`. Will check it out.
I'm going to close the PR.